onnxruntime/core/providers/openvino/openvino_provider_factory.cc

```diff
@@ -83,23 +86,55 @@ struct OpenVINO_Provider : Provider {
     if (provider_options_map.find("device_type") != provider_options_map.end()) {
       device_type = provider_options_map.at("device_type").c_str();
 
-      std::set<std::string> ov_supported_device_types = {"CPU_FP32", "CPU_FP16", "GPU_FP32",
-                                                         "GPU.0_FP32", "GPU.1_FP32", "GPU_FP16",
-                                                         "GPU.0_FP16", "GPU.1_FP16", "NPU"};
+      std::set<std::string> ov_supported_device_types = {"CPU", "GPU",
+                                                         "GPU.0", "GPU.1", "NPU"};
+      std::set<std::string> deprecated_device_types = {"CPU_FP32", "GPU_FP32",
+                                                       "GPU.0_FP32", "GPU.1_FP32", "GPU_FP16",
+                                                       "GPU.0_FP16", "GPU.1_FP16"};
+      if (deprecated_device_types.find(device_type) != deprecated_device_types.end()) {
+        std::string deprecated_device = device_type;
+        int delimit = device_type.find("_");
+        device_type = deprecated_device.substr(0, delimit);
+        precision = deprecated_device.substr(delimit + 1);
```

Comment:
I think we should get a depeacation notice on device type selected to cpu_fp32, gpu_fp32 rather than completely removing the option

Reply:
The option for adding the device_type with precision still exists with depreciation notice.